### ISSUE TITLE
AMPLIFY CLI SET TIME OUT FOR LAMBDA FUNCTIONS

### ISSUE THREAD

**Issue by sacrampton** [aws-amplify/amplify-cli#4164]
**Which Category is your question related to?**

Setting the Time Out for Lambda Functions via the CLI. I would like to also set time out for AppSync, but I don't believe you can do that at this time.

**Amplify CLI Version** 

4.18.1

**What AWS Services are you utilizing?**

AppSync, DynamoDB, ElasticSearch

**Provide additional details e.g. code snippets**

I want to increase the time out period for Lambda functions and am unsure where I should set this in the Amplify project. I think you set something like this, but am not sure and not sure where it should be set.
```
    "Globals":{
        "Function": {"Timeout": 300}
    }
```

**Comment by yuth** [comment]
Do you want to change the timeout of the lambda function itself or do you want to change the timeout for AppSync resolver which uses lambda function

**Comment by sacrampton** [comment]
I'd like to be able to do both if possible, but I didn't think we could set time outs for resolvers.

**Comment by yuth** [comment]
Timeout for AppSync resolvers can't be changed as those are controlled by the AppSync service. Its limited to 30 seconds as listed in their [limitation page](https://docs.aws.amazon.com/general/latest/gr/appsync.html)

For functions you will have to edit the lambda functions cloudformation to change the Timeout parameter. I am going to mark this as an enhancement so we can expose the timeout

**Comment by gadavis2** [comment]
+1

**Comment by maheel** [comment]
@yuth Has this been implemented yet? I don't think it is a good idea to edit the cloudformation file as it will get overwritten if the configurations are changed. Moreover, having an option to change a lambda timeout should be a must-have thing in my view.

**Comment by bart** [comment]
Same issue here but Lambda console shows a default timeout of 25 seconds. Having same issue anyways (same timing difference of 5 seconds in Cloudwatch). Any idea which timeout might get triggered here?

**Comment by VincentGaoHJ** [comment]
Also waiting for this feature to be exposed.

**Comment by markspeich** [comment]
Bump!

**Comment by ChrisJamesC** [comment]
I was able to change the timeout in the <function_name>-awscloudformation-template.json file in the lambda's directory. 

To do so, I added the following line to the "Resources" -> "Lambda Function" field (for 2 minutes): 

    "Timeout": 120

Full documentation can be found here https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html

**Comment by rZinnatov** [comment]
thanks @ChrisJamesC ! just adding my 2 cents :wink: 

**tl;dr for the [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html)**, timeout is located here in the `<function_name>-awscloudformation-template.json` file
```
{
  "Resources": {
    "LambdaFunction": {
      "Properties": {
        "Timeout": 120
      }
    }
  }
}
```
